Add support for OpenBSD systems which have TCP_INFO implemented.
463571f25e8bcb49ae37fec77d63c3fe65dd0638

Index: src/tcp_info.c
--- src/tcp_info.c.orig
+++ src/tcp_info.c
@@ -60,7 +60,7 @@
 int
 has_tcpinfo(void)
 {
-#if (defined(linux) || defined(__FreeBSD__) || defined(__NetBSD__)) \
+#if (defined(linux) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)) \
 	&& defined(TCP_INFO)
     return 1;
 #else
@@ -82,7 +82,7 @@ has_tcpinfo_retransmits(void)
 #else
 #if defined(__FreeBSD__) && __FreeBSD_version >= 600000
     return 1; /* Should work now */
-#elif defined(__NetBSD__) && defined(TCP_INFO)
+#elif (defined(__NetBSD__) || defined(__OpenBSD__)) && defined(TCP_INFO)
     return 1;
 #else
     return 0;
@@ -94,7 +94,7 @@ has_tcpinfo_retransmits(void)
 void
 save_tcpinfo(struct iperf_stream *sp, struct iperf_interval_results *irp)
 {
-#if (defined(linux) || defined(__FreeBSD__) || defined(__NetBSD__)) && \
+#if (defined(linux) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)) && \
 	defined(TCP_INFO)
     socklen_t tcp_info_length = sizeof(struct tcp_info);
 
@@ -118,7 +118,7 @@ get_total_retransmits(struct iperf_interval_results *i
     return irp->tcpInfo.tcpi_total_retrans;
 #elif defined(__FreeBSD__) && __FreeBSD_version >= 600000
     return irp->tcpInfo.tcpi_snd_rexmitpack;
-#elif defined(__NetBSD__) && defined(TCP_INFO)
+#elif (defined(__NetBSD__) || defined(__OpenBSD__)) && defined(TCP_INFO)
     return irp->tcpInfo.tcpi_snd_rexmitpack;
 #else
     return -1;
@@ -138,6 +138,8 @@ get_snd_cwnd(struct iperf_interval_results *irp)
     return irp->tcpInfo.tcpi_snd_cwnd;
 #elif defined(__NetBSD__) && defined(TCP_INFO)
     return (long)irp->tcpInfo.tcpi_snd_cwnd * irp->tcpInfo.tcpi_snd_mss;
+#elif defined(__OpenBSD__) && defined(TCP_INFO)
+    return irp->tcpInfo.tcpi_snd_cwnd;
 #else
     return -1;
 #endif
@@ -158,6 +160,8 @@ get_snd_wnd(struct iperf_interval_results *irp)
     return irp->tcpInfo.tcpi_snd_wnd;
 #elif defined(__NetBSD__) && defined(TCP_INFO)
     return (long)irp->tcpInfo.tcpi_snd_wnd * irp->tcpInfo.tcpi_snd_mss;
+#elif defined(__OpenBSD__) && defined(TCP_INFO)
+    return irp->tcpInfo.tcpi_snd_wnd;
 #else
     return -1;
 #endif
@@ -174,7 +178,7 @@ get_rtt(struct iperf_interval_results *irp)
     return irp->tcpInfo.tcpi_rtt;
 #elif defined(__FreeBSD__) && __FreeBSD_version >= 600000
     return irp->tcpInfo.tcpi_rtt;
-#elif defined(__NetBSD__) && defined(TCP_INFO)
+#elif (defined(__NetBSD__) || defined(__OpenBSD__)) && defined(TCP_INFO)
     return irp->tcpInfo.tcpi_rtt;
 #else
     return -1;
@@ -192,7 +196,7 @@ get_rttvar(struct iperf_interval_results *irp)
     return irp->tcpInfo.tcpi_rttvar;
 #elif defined(__FreeBSD__) && __FreeBSD_version >= 600000
     return irp->tcpInfo.tcpi_rttvar;
-#elif defined(__NetBSD__) && defined(TCP_INFO)
+#elif (defined(__NetBSD__) || defined(__OpenBSD__)) && defined(TCP_INFO)
     return irp->tcpInfo.tcpi_rttvar;
 #else
     return -1;
@@ -223,11 +227,7 @@ build_tcpinfo_message(struct iperf_interval_results *r
 	    r->tcpInfo.tcpi_lost, r->tcpInfo.tcpi_retrans, r->tcpInfo.tcpi_fackets,
 	    r->tcpInfo.tcpi_rtt, r->tcpInfo.tcpi_reordering);
 #endif
-#if defined(__FreeBSD__) && defined(TCP_INFO)
-    sprintf(message, report_tcpInfo, r->tcpInfo.tcpi_snd_cwnd,
-	    r->tcpInfo.tcpi_rcv_space, r->tcpInfo.tcpi_snd_ssthresh, r->tcpInfo.tcpi_rtt);
-#endif
-#if defined(__NetBSD__) && defined(TCP_INFO)
+#if (defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)) && defined(TCP_INFO)
     sprintf(message, report_tcpInfo, r->tcpInfo.tcpi_snd_cwnd,
 	    r->tcpInfo.tcpi_rcv_space, r->tcpInfo.tcpi_snd_ssthresh, r->tcpInfo.tcpi_rtt);
 #endif
